### PR TITLE
Add missing/negations type assertions/selectors

### PR DESCRIPTION
**Pull request by jnyrup**
Adds lot `Not` Assertions/Selectors for types/methods/properties/.

The deletions are:
* Renaming of existing tests to fix ambiguity conflict with new tests of Not variant.
* double white space typo
* typo in summary for ThatAreUnderNamespace

* [ ] The [Pull Request](https://help.github.com/articles/using-pull-requests) is targeted at the `master` branch.
* [x] The code complies with the [Coding Guidelines for C# 3.0, 4.0 and 5.0](http://www.csharpcodingguidelines.com/)/.
* [x] The changes are covered by a new or existing set of unit tests which follow the Arrange-Act-Assert syntax such as is used [in this example](https://github.com/fluentassertions/fluentassertions/blob/daaf35b9b59b622c96d0c034e8972a020b2bee55/Tests/FluentAssertions.Shared.Specs/BasicEquivalencySpecs.cs#L33).
* [x] If the contribution affects the documentation, please include your changes to [**documentation.md**](https://github.com/fluentassertions/fluentassertions/blob/master/docs/documentation.md) in this pull request so the documentation will appear on the [website](http://fluentassertions.com/documentation.html).

### CODE REVIEW

**Comment by dennisdoomen** [comment]
Awesome work @jnyrup. How can I contact you?

**Comment by jnyrup** [comment]
If you mean by e.g email I have a gmail. It's the same as my github username.

File: Src/FluentAssertions/Types/MethodBaseAssertions.cs [hunk 1]
```diff
@@ -37,6 +37,28 @@ public abstract class MethodBaseAssertions<TSubject, TAssertions> : MemberInfoAs
...
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion 
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <see cref="because" />.
+        /// </param>
+        public AndConstraint<TAssertions> NotHaveAccessModifier(
+            CSharpAccessModifier accessModifier, string because = "", params object[] becauseArgs)
+        {
+            Execute.Assertion.ForCondition(accessModifier != Subject.GetCSharpAccessModifier())
```
Comment (dennisdoomen) [inline Src/FluentAssertions/Types/MethodBaseAssertions.cs]: Purely for readability, I prefer the `ForCondition` on the next line

File: Src/FluentAssertions/Types/MethodInfoAssertions.cs [hunk 2]
```diff
@@ -44,6 +44,31 @@ public MethodInfoAssertions(MethodInfo methodInfo)
...
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion 
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <see cref="because" />.
+        /// </param>
+        public AndConstraint<MethodInfoAssertions> NotBeVirtual(
+            string because = "", params object[] becauseArgs)
+        {
+            string failureMessage = "Expected method " +
```
Comment (dennisdoomen) [inline Src/FluentAssertions/Types/MethodInfoAssertions.cs]: Why not put this on a single line, preferably using string interpolation? Same for similar uses elsewhere in this PR.

File: Src/FluentAssertions/Types/MethodInfoAssertions.cs [hunk 3]
```diff
@@ -64,6 +89,26 @@ public AndConstraint<MethodInfoAssertions> BeAsync(string because = "", params o
...
 
         /// <summary>
+        /// Asserts that the selected method is not async. 
+        /// </summary>
+        /// <param name="because">A formatted phrase as is supported by <see cref="M:System.String.Format(System.String,System.Object[])"/> explaining why the assertion
+        ///             is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.</param>
+        /// <param name="becauseArgs">Zero or more objects to format using the placeholders in <see cref="!:because"/>.</param>
+        public AndConstraint<MethodInfoAssertions> NotBeAsync(string because = "", params object[] becauseArgs)
+        {
+            string failureMessage = "Expected subject " +
```
Comment (dennisdoomen) [inline Src/FluentAssertions/Types/MethodInfoAssertions.cs]: Should probably say *method* rather than *subject*

**Comment by jnyrup** [comment]
While I feel the expression "expected something to not be" is more strict, I do prefer your suggestion.
Without being an expert (at all) in the English language your suggestion does sound more correct.
It also makes it easier to distinguish a `Be` and `NotBe` as the negation is in the beginning of the message instead of in the middle somewhere.

**Comment by jnyrup** [comment]
Now I'm a bit unsure. 

There are around 124 other places that uses the construct _Expected something to not be something_ or _Expected something not to be something_.
E.g `Expected {context:collection} not to be empty{reason}.`, which does become weaker if changing it to 
`Did not expect {context:collection} to be empty{reason}.` Now it sounds as if we didn't explicitly asserted it not to be empty.

I quick guess is that all instances of _Expected something to not be something_ should be changed to _Expected something not to be something_?

**Comment by dennisdoomen** [comment]
OK, just ignore that comment.

**Pull request merged**